crates/bindings-csharp/Runtime/bindings.c

```diff
@@ -104,20 +105,26 @@ PREINIT(10, startup) {
     MonoObject* res;                                                          \
     mono_wasm_invoke_method_ref(ffi_method_##name, NULL, (void*[]){args},     \
                                 NULL, &res);                                  \
-    return *(ret*)mono_object_unbox(res);                                     \
+    res_code                                                                  \
   }
 
-EXPORT(Buffer, __describe_module__, ());
+#define EXPORT(ret, name, params, args...)                                             \
+  EXPORT_WITH_MONO_RES(ret, return *(ret*)mono_object_unbox(res);, name, params, args) \

```

Comment:
God I can't wait for when we'll be able to ditch all this Mono stuff. .NET 9, I'm counting on you...